lib/logstash/inputs/log4j.rb

```diff
@@ -82,12 +98,27 @@ def create_event(log4j_obj)
     return event
   end # def create_event
 
+  # method used to stop the plugin and unblock
+  # pending blocking operatings like sockets and others.
+  def stop
+    super
+    @server_socket.close if @server_socket
```

Comment:
We could probably do a check if the socket is already closed?

```
@server_socker.close if @server_socket && !@server_socket.closed?
```

If its already close the `#close` method will raise an `IOError: closed stream`